src/AttendanceHandler.hpp

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include "Lesson.hpp"
+#include "User.hpp"
+#include "Attendance.hpp"
+
+#include <unordered_map>
+
+class AttendanceHandler
+{
+private:
+  auto matchByEmail(const User& userProfile);
+
+public:
+  using AttendanceList = std::unordered_multimap<std::string, Attendance>;
+
+  void reportAbsence(const User& userProfile,
+                     const Lesson& Lesson, 
+                     const std::string& comment);
+
+  void reportMakeUp(const User& userProfile,
+                    const Lesson& Lesson,
+                    const std::string& absenceDate,
+                    const std::string& comment);
+
+
+  AttendanceList viewReports(const User * const userProfile);
```

Comment:
why not smart_ptr? the name viewReports is IMO not relevant. It just returns an AttendanceList, not display it. `getReports()`?

Reply:
Name changed

Reply:
My reason was: I need a parameter that can be a User and a nullptr.
But I will not change the object, I will not take ownership of the object and I will not reassign it.

so I need something like a reference but with ability to store "nullptr". 

Smart_ptr look for as too much functionality for my needs so I chose const User * const

but as we use c++ I change const User* const to shared_ptr.